### PR TITLE
Issue #52:  Prevent including fields without columns in Headless

### PR DESCRIPTION
**Pull request by valenvivaldi**
ETP-2271

### CODE REVIEW

File: src/com/etendoerp/etendorx/openapi/DynamicDatasourceEndpoint.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public class DynamicDatasourceEndpoint implements OpenAPIEndpoint {
 
   private ThreadLocal<String> requestedTag = new ThreadLocal<>();
 
-  private static final List<String> extraFields = List.of("_identifier", "$ref", "active", "creationDate", "createdBy",
+  static final List<String> extraFields = List.of("_identifier", "$ref", "active", "creationDate", "createdBy",
       "createdBy$_identifier", "updated", "updatedBy", "updatedBy$_identifier");
```
Comment (etendobot) [inline src/com/etendoerp/etendorx/openapi/DynamicDatasourceEndpoint.java:63]: **Suggestion**

Consider making extraFields private to better encapsulate internals and adhere to typical Etendo visibility practices, unless package or test access is strictly required.

```java
private static final List<String> extraFields = List.of("_identifier", "$ref", "active", "creationDate", "createdBy",
    "createdBy$_identifier", "updated", "updatedBy", "updatedBy$_identifier");
```

**Rationale**: Encapsulation is a widely followed pattern in Etendo, and unnecessary exposure of internals can risk accidental coupling or misuse.

File: src-test/src/com/etendoerp/etendorx/openapi/DynamicDatasourceEndpointTest.java [hunk 2]
```diff
@@ -275,4 +276,49 @@ public void testDefineResponseSchema() {
     assertNotNull(schema.getProperties().get("status"));
     assertNotNull(schema.getProperties().get("data"));
   }
+
+  @Test
+  public void testGetRequestBody_SkipsNullColumnsAndAddsExtraFields() throws Throwable {
```
Comment (etendobot) [inline src-test/src/com/etendoerp/etendorx/openapi/DynamicDatasourceEndpointTest.java:281]: **Suggestion**

Consider documenting the edge case being tested—namely, that fields with null columns are skipped but extra fields are still included. While this is a test, clarifying commentary aids future maintainability.

```java
@Test
  public void testGetRequestBody_SkipsNullColumnsAndAddsExtraFields() throws Throwable {
```

**Rationale**: Etendo test case documentation often includes a brief description of edge case and intent, improving long-term test clarity.

File: src-test/src/com/etendoerp/etendorx/services/DataSourceServletTest.java [hunk 3]
```diff
@@ -196,4 +198,112 @@ public int read() throws IOException {
     JSONObject data = result.getJSONObject(DataSourceConstants.DATA);
     assertEquals("value1", data.getString("field1"));
   }
+
+  @Test
+  public void testGetTab_IgnoresFieldsWithNullColumnAndOrdersBySeqNo() throws Throwable {
```
Comment (etendobot) [inline src-test/src/com/etendoerp/etendorx/services/DataSourceServletTest.java:203]: **Suggestion**

Great job covering field column null exclusion and sequence ordering edge cases with comprehensive mocking.

```java
@Test
  public void testGetTab_IgnoresFieldsWithNullColumnAndOrdersBySeqNo() throws Throwable {
```

**Rationale**: Extra tests on field nullability and sequence provide strong regression coverage for DataSourceServlet field-handling logic, consistent with Etendo test standards.

**Review by etendobot — APPROVED** [review]
